lib/fastlane/actions/appium.rb

```diff
@@ -0,0 +1,145 @@
+module Fastlane
+  module Actions
+    class AppiumAction < Action
+      INVOKE_TIMEOUT = 30
+
+      def self.run(params)
+        require 'rspec'
+        require 'appium_lib'
+
+        Helper.log.info "Parameter Platform: #{params[:platform]}"
+        Helper.log.info "Parameter Spec Path: #{params[:spec_path]}"
+        Helper.log.info "Parameter App Path: #{params[:app_path]}"
+        Helper.log.info "Parameter Invoke Appium Server: #{params[:invoke_appium_server]}"
+        Helper.log.info "Parameter Appium Server Host: #{params[:host]}"
+        Helper.log.info "Parameter Appium Server Port: #{params[:port]}"
+        Helper.log.info "Parameter Appium Capabilities: #{params[:caps]}"
+
+        if params[:invoke_appium_server]
+          appium_pid = invoke_appium_server(params)
+          wait_for_appium_server(params)
+        end
+
+        configure_rspec(params)
+
+        rspec_args = []
+        rspec_args << params[:spec_path]
+        status = RSpec::Core::Runner.run(rspec_args).to_i
+        if status != 0
+          raise "Failed to run appium spec. status code: #{status}".red
+        end
+      ensure
+        Actions.sh "kill #{appium_pid}" if appium_pid
+      end
+
+      def self.invoke_appium_server(params)
+        if `which appium`.to_s.length == 0
+          raise 'You have to install appium using `npm install -g appium`'.red
+        end
+
+        fork do
+          Process.exec("appium -a #{params[:host]} -p #{params[:port]}")
```

Comment:
`appium` command invokes HTTP server process.
Appium server is not killed at action finished when I use `sh`.
I would like to kill `appium` process at action finished for idempotence.

Do you have any ideas?